CMakeLists.txt

```diff
@@ -68,6 +68,10 @@ if(NOT CMAKE_CXX_STANDARD)
 endif()
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# Explicitly enable coroutine support, since GCC does not enable it
+# by default when targeting C++17.
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fcoroutines>)
```

Comment:
We can limit this to GCC and exclude other compilers that may not need this or that may not support this feature switch.

Reply:
Done!